pyccel/parser/semantic.py

```diff
@@ -2699,6 +2699,12 @@ def _handle_is_operator(self, IsClass, expr, **settings):
         var1 = self._visit(expr.lhs)
         var2 = self._visit(expr.rhs)
 
+        if (var1 == var2):
```

Comment:
This should only compare Variable instances. I think this will return `True` for the following case when it should raise an error:
```
In [7]: a+b is a+b
Out[7]: False
```

Reply:
I have just run the test and this is indeed translated to `True`

Reply:
yes you are right

Reply:
@EmilyBourne Good catch!